tensorboard/components/plugin_util/test/plugin-test.ts

```diff
@@ -175,6 +179,45 @@ namespace tf_plugin.test {
 
           expect(barCb.callCount).to.equal(1);
         });
+
+        it('ignores foreign postMessages', async function() {
+          const barCb = this.sandbox.stub();
+          this.destListen('bar', barCb);
+          const fakeMessage: Message = {
+            type: 'bar',
+            id: 0,
+            payload: '',
+            error: null,
+            isReply: false,
+          };
+          this.destWindow.postMessage(JSON.stringify(fakeMessage), '*');
+
+          // Await another message to ensure fake message was handled in dest.
+          await this.srcSendMessage('not-bar');
+          expect(barCb.callCount).to.equal(0);
+        });
+
+        it('processes messages while waiting for a reponse', async function() {
+          let resolveLongTask = null;
+          this.destListen('longTask', () => {
+            return new Promise((resolve) => {
+              resolveLongTask = resolve;
+            });
+          });
+
+          const longTaskCb = this.sandbox.stub();
```

Comment:
suffix `Stub` (instead of Cb) may be more legible as it is clear that you are using sinon stub here.

Reply:
Hmm, Done